airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -508,10 +512,15 @@ def _merge_stream_slicers(self, model: DeclarativeStreamModel, config: Config) -
                 else self._create_component_from_model(model=stream_slicer_model, config=config)
             )
 
-        if incremental_sync and stream_slicer:
-            return CartesianProductStreamSlicer(stream_slicers=[incremental_sync, stream_slicer], parameters=model.parameters)
-        elif incremental_sync:
-            return incremental_sync
+        if model.incremental_sync and stream_slicer:
```

Comment:
I need to validate this behavior with custom component. As is, I'm almost certain this would break their state since it would migrate it to the new state per partition. This might mean that:
* https://github.com/airbytehq/airbyte/issues/27024 is needed to merge this
I would also like to review the sources in our repository to make sure there are no other sources than zenloop that has incremental sync and partition routing